bot/plotting/random_plot_generator.py

```diff
@@ -95,6 +113,28 @@ def random_plot_generator(
                     "SEI": sei,
                 })
 
+            solvers = [
+                pybamm.CasadiSolver(mode="safe"),
+                pybamm.CasadiSolver(mode="fast with events")
+            ]
+
+            # choose a random solver
+            solver = random.choice(solvers)
+
+            # 0: pre-defined model with a pre-defined chemistry
+            # 1: experiment with summary variable
+            # 2: experiment without summary variables
+            # 3: comparison plots
+            choice = random.randint(0, 3)
+
+            # if testing, don't randomly choose stuff
+            if testing is True and provided_choice is not None:
+                choice = provided_choice
+
+            # Add degradation only if we are plotting summary variables
```

Comment:
you could move setting degradation options to here so that you don't necessarily set the options and then overwrite them